services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -788,11 +788,40 @@ func TestReleaseTrainErrors(t *testing.T) {
 }
 
 func TestRbacTransformerTest(t *testing.T) {
+	envGroupProduction := "production"
 	tcs := []struct {
 		Name          string
 		Transformers  []Transformer
 		ExpectedError string
 	}{
+		{Name: "able to create environment with permissions policy",
+			Transformers: []Transformer{
+				&CreateEnvironment{
+					Environment: "production-p1",
+					Authentication: Authentication{RBACConfig: auth.RBACConfig{DexEnabled: true, Policy: map[string]*auth.Permission{
+						"developer,CreateEnvironment,*:*,*,allow": {Role: "developer"}}}}},
+			},
+		},
+		{
+			Name: "able to create environment inside environment group with permissions policy",
+			Transformers: []Transformer{
+				&CreateEnvironment{
+					Environment: "production-p2",
+					Config:      config.EnvironmentConfig{EnvironmentGroup: &envGroupProduction},
+					Authentication: Authentication{RBACConfig: auth.RBACConfig{DexEnabled: true, Policy: map[string]*auth.Permission{
+						"developer,CreateEnvironment,production:*,*,allow": {Role: "developer"}}}}},
+			},
+		},
+		{
+			Name: "unable to create environment without permissions policy",
+			Transformers: []Transformer{
+				&CreateEnvironment{
+					Environment:    "production-p2",
+					Config:         config.EnvironmentConfig{EnvironmentGroup: &envGroupProduction},
+					Authentication: Authentication{RBACConfig: auth.RBACConfig{DexEnabled: true, Policy: map[string]*auth.Permission{}}}},
+			},
+			ExpectedError: "user does not have permissions for: developer,CreateEnvironment,production:*,*,allow",
```

Comment:
```suggestion
			ExpectedError: "user (username) does not have permissions for: developer,CreateEnvironment,production:*,*,allow",
```

Reply:
Would be great if we can give out the current user name or email

Reply:
I agree that this is useful, however, let me do this on a separate PR to avoid any confusion.